### ISSUE TITLE
datamodel-codegen creates models with duplicate fields

### ISSUE THREAD

**Issue by pritishperpetua** [koxudaxi/datamodel-code-generator#763]
**Describe the bug**

- I am trying to auto-generate schema models using datamodel-codegen on an openapi documentation:- https://d3a0d0y2hgofx6.cloudfront.net/openapi/en-us/dsp/3-1/openapi.yaml
- The generated schema models has duplicate fields with incorrect typehints. 

**To Reproduce**

Used command line:
```
$ curl https://d3a0d0y2hgofx6.cloudfront.net/openapi/en-us/dsp/3-1/openapi.yaml -o amz_openapi.yaml
$ swagger-cli bundle --dereference amz_openapi.yaml > amz_openapi_dereferenced.yaml
$ datamodel-codegen --reuse-model --target-python-version 3.9 --use-default --input amz_openapi_dereferenced.yaml  --output amz_api_schema.py
```

**Expected behavior**
- Model `OrderV22` has `currencyCode` as a string field. 

**Observed behavior**
- Model `OrderV22` has multiple fields for `currencyCode`

<img width="1768" alt="image" src="https://user-images.githubusercontent.com/61429653/167482207-834022c8-62b3-46d6-8f56-8193bd89e667.png">

Similar behavior can also be observed for `LineItemV31`

Any insights on what might me going wrong here would be really appreciated. 

**Version:**
 - OS: macOS
 - Python version: 3.9
 - datamodel-code-generator version: 0.12.0

**Comment by koxudaxi** [comment]
@pritishperpetua  
I have released the PR as `0.15.0`
Thank you very much!!

**Comment by dannyrohde** [comment]
I believe this change introduces a problem for generating a Pedantic model from an Amazon JSON source.

In the past, the library generated a list of properties on the class, with the *last* one being correct:

```
    fulfillment_availability: Optional[List[Any]] = None
    fulfillment_availability: Optional[List[Any]] = None
    fulfillment_availability: Optional[List[Any]] = None
    fulfillment_availability: Optional[List[Any]] = None
    fulfillment_availability: Optional[List[Any]] = None
    fulfillment_availability: Optional[List[Any]] = None
[…]
    fulfillment_availability: Optional[List[FulfillmentAvailabilityItem]] = Field([…])
```

I'm unsure where the initial declarations come from as the JSON is quite complex, and I don't fully understand it. However, given the correct declaration was listed last, the class worked as expected. With the change in 0.15 above, the code will pick the first instance and skip over any subsequent instance, leaving the class with an incorrectly parsed property.

It's not clear to me how to solve that elegantly, as the initial bug report seems to have had an issue with later declarations (the first one being correct), whereby I seem to have the opposite situation (the last declaration being correct).

Do you have an idea how to deal with this? Another settings switch to take the first/last occurrence seems clumsy. Any other suggestions?

Thanks
